src/wrappers/panel.js

```diff
@@ -0,0 +1,167 @@
+import { getContainerNode, renderTemplate } from '../lib/utils';
+import { component } from '../lib/suit.js';
+import cx from 'classnames';
+import defaultTemplates from './defaultTemplates';
+
+export default widgetFactory => optionsWithPanelOpts => {
+  const {
+    container,
+    cssClasses,
+    templates = {},
+    collapsible,
+  } = optionsWithPanelOpts;
+
+  const allTemplates = {
+    ...defaultTemplates,
+    ...templates,
+  };
+
+  const panel = renderPanel({
+    container: getContainerNode(container),
+    cssClasses,
+    templates: allTemplates,
+    collapsible,
+  });
+
+  const widget = widgetFactory({
+    ...optionsWithPanelOpts,
+    container: panel.body,
+    templates: {
+      ...optionsWithPanelOpts.templates,
+      header: undefined,
+      footer: undefined,
+      collapseButton: undefined,
+    },
+  });
+
+  return {
+    getConfiguration: opts =>
+      widget.getConfiguration ? widget.getConfiguration(opts) : {},
+    init: opts => {
+      widget.init(opts);
+    },
```

Comment:
I didn't think about that. So yes it would remove the repetition and it will make the implementation more robust. What about using the prototype instead of copying the methods?

Reply:
> What about using the prototype instead of copying the methods?

What do you mean by that?

Reply:
```js
const wrappedWidget = Object.create(widget);
wrappedWidget.render = args => {
  widget.render(args);
  // ... other stuff
}
```

This way we rely on the prototype chain for dispatching the calls to the methods.

Reply:
I don't think we need to rely on the prototype. IMO the `...widget` is enough.

Reply:
There might be something I don't see, why is that prototype chain an issue?

Reply:
There is no issue. It's just that we don't leverage the power of the prototype chain in the codebase. Since we don't create instance we don't share the prototype with other widgets. That's why I think we can stick with an object, but it also works with `Object.create`.

Reply:
OK let's go for `...` then 👍